src/components/Buttons/AddFolderOptionBtn.tsx

```diff
@@ -0,0 +1,81 @@
+import styled from "styled-components";
+import getFolderTagListData from "../../utils/getFolderTagListData";
+import { useEffect, useState } from "react";
+import AddLinkToFolder from "../../modals/contents/AddLinkToFolder";
+import DeleteLink from "../../modals/contents/DeleteLink";
+import { IFolderTagData } from "../../utils/types/common.types";
+
+const OptionBtnStyle = styled.button`
+  width: 100px;
+  padding: 7px 12px;
+  color: #333236;
+  border: none;
+  background: #fff;
+  box-shadow: 0px 2px 8px 0px rgba(51, 50, 54, 0.1);
+  &:active {
+    background: #e7effb;
+    color: #6d6afe;
+  }
+`;
+
+const OptionBtnContainer = styled.div<{ isActive: boolean }>`
+  display: flex;
+  flex-direction: column;
+  gap: 2px;
+  position: absolute;
+  bottom: 6px;
+  left: 108px;
+  visibility: ${({ isActive }) => (isActive ? "visible" : "hidden")};
+`;
+
+export interface IAddFolderOptionBtnProps {
+  options: string[];
+  isFloatingBtnActive: boolean;
+}
+
+function AddFolderOptionBtn({
+  options,
+  isFloatingBtnActive = false,
+}: IAddFolderOptionBtnProps) {
+  const [isDeleteOpen, setDeleteOpen] = useState<boolean>(false);
+  const [isAddLinkOpen, setAddLinkOpen] = useState<boolean>(false);
+  const [isActive, setIsActive] = useState<boolean>(isFloatingBtnActive);
+  const handleClick = (selectedOption: string) => {
+    if (selectedOption === "삭제하기") setDeleteOpen(true);
+    else if (selectedOption === "폴더에 추가") setAddLinkOpen(true);
+    setIsActive(false);
+  };
+  const changeDeleteOpenState = (openState: boolean) =>
+    setDeleteOpen(openState);
+  const changeAddLinkOpenState = (openState: boolean) =>
+    setAddLinkOpen(openState);
+  const TagDataList: IFolderTagData[] = getFolderTagListData();
+
+  useEffect(() => {
+    setIsActive((prev) => !prev);
+  }, [isFloatingBtnActive]);
+
+  return (
+    <div>
+      <OptionBtnContainer isActive={isActive}>
+        {options &&
```

Comment:
options는 props의 타입을 보면 `string[]` 로, optional value가 아니죠?
그러면 이렇게 다시 options가 있을 경우로만 예외 처리 해주실 필요 없어요.
만약 options가 상황에 따라 있을수도있고, 없을수도 있는 value라면 `IAddFolderOptionBtnProps` 에서도 이렇게 작성해주세요.

```tsx
export interface IAddFolderOptionBtnProps {
  options?: string[];
  ...
}
```

Reply:
그리고 optional chaining 쓰시는게 문법이 더 간결해보일거예요.
```suggestion
        options?.map((optionText) => (
```